File: demo/src/main/java/hu/telekom/ubuntureleasedemo/java/UbuntuReleaseClient.java
package hu.telekom.ubuntureleasedemo.java;

import org.springframework.stereotype.Component;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Mono;

@Component
public class UbuntuReleaseClient {

    private WebClient ubuntuReleaseWebClient;

    public UbuntuReleaseClient(WebClient client) {
        this.ubuntuReleaseWebClient = client;
    }

    public Mono<String> getVersion(int idx) {
        return ubuntuReleaseWebClient
            .get().uri("/version/" + idx).retrieve()
            .bodyToMono(String.class);
    }

    public Mono<String> getFirstCodeName(String version) {
        return ubuntuReleaseWebClient
            .get().uri("/codename/first/" + version).retrieve()
            .bodyToMono(String.class);
    }

    public Mono<String> getLastCodeName(String version) {
        return ubuntuReleaseWebClient
            .get().uri("/codename/last/" + version).retrieve()
            .bodyToMono(String.class);
    }

    public Mono<String> getLts(String codename) {
        return ubuntuReleaseWebClient
            .get().uri("/lts/" + codename).retrieve()
            .bodyToMono(String.class);
    }

}

File: demo/src/main/kotlin/UbuntuReleaseDemo/UbuntuReleaseClient.kt
package hu.telekom.ubuntureleasedemo.kotlin

import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.client.WebClient
import org.springframework.web.reactive.function.client.awaitBody

@Component
class UbuntuReleaseClient(private val ubuntuReleaseWebClient: WebClient) {

    suspend fun getVersion(idx: Int): String = ubuntuReleaseWebClient
            .get().uri("/version/$idx").retrieve()
            .awaitBody()

    suspend fun getFirstCodeName(version: String): String = ubuntuReleaseWebClient
            .get().uri("/codename/first/$version").retrieve()
            .awaitBody()

    suspend fun getLastCodeName(version: String): String = ubuntuReleaseWebClient
            .get().uri("/codename/last/$version").retrieve()
            .awaitBody()

    suspend fun getLts(codename: String): String = ubuntuReleaseWebClient
            .get().uri("/lts/$codename").retrieve()
            .awaitBody()

}
